src/vm/hints/usort.rs

```diff
@@ -0,0 +1,227 @@
+        "usort_max_size".to_string(),
+        usort_max_size,
+    )]));
+    Ok(())
+}
+
+pub fn usort_body(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let _ = get_range_check_builtin(vm)?;
+    let input_arr_start_ptr = get_relocatable_from_var_name("input", ids, vm, hint_ap_tracking)?;
+    let input_ptr = vm.memory.get_relocatable(&input_arr_start_ptr)?.clone();
+    let usort_max_size = get_int_from_scope(vm, "usort_max_size");
+    let input_len = get_integer_from_var_name("input_len", ids, vm, hint_ap_tracking)?;
+    let input_len_usize = input_len
+        .to_usize()
+        .ok_or(VirtualMachineError::BigintToUsizeFail)?;
+
+    if let Some(usort_max_size) = usort_max_size {
+        if input_len > &usort_max_size {
+            return Err(VirtualMachineError::UsortOutOfRange(
+                usort_max_size,
+                input_len.clone(),
+            ));
+        }
+    }
+
+    let mut positions_dict: HashMap<BigInt, Vec<BigInt>> = HashMap::new();
+    let mut output: Vec<BigInt> = Vec::new();
+    for i in 0..input_len_usize {
+        let val = get_integer_from_relocatable_plus_offset(&input_ptr, i, vm)?;
+        if let Err(output_index) = output.binary_search(val) {
+            output.insert(output_index, val.clone());
+        }
+        positions_dict
+            .entry(val.clone())
+            .or_insert(Vec::new())
+            .push(bigintusize!(i));
+    }
+
+    let mut multiplicities: Vec<BigInt> = Vec::new();
+    for k in output.iter() {
+        multiplicities.push(bigintusize!(positions_dict[k].len()));
+    }
+
+    vm.exec_scopes
+        .assign_or_update_variable("positions_dict", PyValueType::KeyToListMap(positions_dict));
+    let output_base = vm.segments.add(&mut vm.memory, Some(output.len()));
+    let multiplicities_base = vm.segments.add(&mut vm.memory, Some(multiplicities.len()));
+    let output_len = output.len();
+
+    for (i, sorted_element) in output.into_iter().enumerate() {
+        insert_integer_at_relocatable_plus_offset(sorted_element, &output_base, i, vm)?;
+    }
+
+    for (i, repetition_amount) in multiplicities.into_iter().enumerate() {
+        insert_integer_at_relocatable_plus_offset(repetition_amount, &multiplicities_base, i, vm)?;
+    }
+
+    insert_integer_from_var_name(
+        "output_len",
+        bigintusize!(output_len),
+        ids,
+        vm,
+        hint_ap_tracking,
+    )?;
+
+    insert_relocatable_from_var_name("output", output_base, ids, vm, hint_ap_tracking)?;
+
+    insert_relocatable_from_var_name(
+        "multiplicities",
+        multiplicities_base,
+        ids,
+        vm,
+        hint_ap_tracking,
+    )
+}
+
+pub fn verify_usort(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let _ = get_range_check_builtin(vm);
```

Comment:
We don't seem to care about this.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/319/commits/9cabded7e3919383bc148d946c761dd7c9487174